var/spack/repos/builtin/packages/zlib-ng/package.py

```diff
@@ -55,9 +57,14 @@ class ZlibNg(AutotoolsPackage, CMakePackage):
 
     @property
     def libs(self):
-        name = "libz" if self.spec.satisfies("+compat") else "libz-ng"
+        compat_name = "zlib" if sys.platform == "win32" else "libz"
+        name = compat_name if self.spec.satisfies("+compat") else "libz-ng"
```

Comment:
> (question) does Windows support ~compat? I assume if it did, and it built a library, it would not be called libz-ng

Yes, although I have no idea what that library name would be. I'll update it once I run the build. Good catch.

> As a side note, it came up a couple times in this PR so regarding these library names: we might want to make an inherited function that can generate an appropriate platform-specific library-naming format.

I actually considered (and started) making modifications to `llnl.util.filesystem.find_library` and implementing heuristics similar to CMake's `find_` modules, but quickly realized packages on Windows are too inconsistent. Some still do use the `lib` prefix, some do not, etc. Further, most Spack packages already search for `libxyz`, so in addition to the extra logic, we'd also have to manipulate the library names passed in, and even that won't cover all cases. 

I'm happy to improve the `find_library` behavior (seems like a very worthwhile effort IMO) but its sufficiently complicated I don't think it belongs in this PR.

Reply:
The non compat name is just `zlib-ng` probably should have been able to guess that 🤦